### DISCUSSION TITLE
Patch users table returns 404 when sending after user signups

### DISCUSSION THREAD

**Discussion by surjithctly** [supabase/supabase#2201] [Questions]
I have this code where I wait for signup and then using an SQL function, I create the user in "users" table. Now I try to update the user's table and it throws a `404 error`.  

```
const { error, user } = await signUp({ email, password });
if (user) {
        await updateUserName(user, name);
}
```

This is the update function

```
export const updateUserName = async (user, name) => {
  await supabase
    .from("users")
    .update({
      full_name: name,
    })
    .eq("id", user.id);
};
```

Code is taken from https://github.com/vercel/nextjs-subscription-payments but no idea why it's not working locally.  I have also taken the new_user_table function from here: https://github.com/supabase/supabase/discussions/306

Any help would be appreciated.

**Comment by kangmingtay** [comment]
Hi @surjithctly, if I'm not mistaken, the `users`  table doesn't have a column called `full_name`. Also, if you wish to update a user's data, it would be better to use the `update` method (https://github.com/supabase/gotrue-js/blob/4575b2a116048b8226b5d4bd12a33777571b9b2a/src/GoTrueClient.ts#L244) which checks if the user is logged in with the correct access token before making the update.

Let me know if this helps! Keep hacking :)
